src/components/views/rooms/RoomHeader.tsx

```diff
@@ -67,9 +72,26 @@ function showOrHidePanel(phase: RightPanelPhases): void {
 }
 
 export default function RoomHeader({ room }: { room: Room }): JSX.Element {
+    const client = useMatrixClientContext();
+
     const roomName = useRoomName(room);
     const roomTopic = useTopic(room);
 
+    const directRoomsList = useAccountData<Record<string, string[]>>(client, EventType.Direct);
+    const [isDirectMessage, setDirectMessage] = useState(false);
+
+    const members = useRoomMembers(room);
+    const memberCount = useRoomMemberCount(room);
+
+    useEffect(() => {
+        for (const [, dmRoomList] of Object.entries(directRoomsList)) {
+            if (dmRoomList.includes(room?.roomId ?? "")) {
+                setDirectMessage(true);
+                break;
+            }
+        }
+    }, [room, directRoomsList]);
```

Comment:
This kind of combination of `useState` and `useEffect` is an anti-pattern:

``` ts
const [state, setState] = useState(0)
useEffect(() => {
  setState(things.length)
}, [things])
```

because it could be simplified to `const state = useMemo(() => things.length, [things])`, which also avoids triggering an unnecessary render. Couldn't `isDirectMessage` be simplified to a `useMemo` as well?

Reply:
you're absolutely right, thank you for the heads up 🙏